prsa/quantification.py

```diff
@@ -0,0 +1,155 @@
+    sum_ac_lower = sum(prsa_output_ac[L - s + 1:L + 1])
+    ac = (1 / (2 * s)) * (sum_ac_upper - sum_ac_lower)
+
+    return ac
+
+def calculate_dc(prsa_output_dc: np.ndarray, L: int, s: int) -> float:
+    """
+    Calculate the Deceleration Capacity (DC) of a signal using phase-rectified signal averaging.
+
+    The formula for DC is given by:
+    DC = (1 / (2s)) * Σ(x_DC[i]) from i = L+1 to L+s - (1 / (2s)) * Σ(x_DC[i]) from i = L-s to L
+    where:
+    - x_DC[i] is the phase-rectified signal for deceleration capacity at point i,
+    - L is the anchor point around which the window is considered,
+    - s is the parameter for summarizing the phase-rectified curves (assumed to be even).
+
+    Parameters:
+    prsa_output_dc (np.ndarray): The input phase-rectified signal.
+    L (int): The anchor point around which the DC is calculated.
+    s (int): The summarizing parameter for the phase-rectified curves.
+
+    Returns:
+    float: The calculated DC value.
+    """
+    L = int(L / 2)
+    sum_dc_upper = sum(prsa_output_dc[L + 1:L + s + 1])
+    sum_dc_lower = sum(prsa_output_dc[L - s:L])
+    dc = (1 / (2 * s)) * (sum_dc_upper - sum_dc_lower)
+
+    return dc
+
+"""
+The formula for calculating the AAC is based on the following article:
+Lobmaier, Silvia & Huhn, Evelyn & Steinburg, S. & Müller, Alexander & Schuster, Tibor & Ortiz,
+Javier & Schmidt, Georg & Schneider, Karl. (2012). 
+Phase-rectified signal averaging as a new method for surveillance of growth restricted fetuses. 
+The journal of maternal-fetal & neonatal medicine : the official journal of the European Association of Perinatal Medicine, 
+the Federation of Asia and Oceania Perinatal Societies, the International Society of Perinatal Obstetricians. 25. 10.3109/14767058.2012.696163. 
+
+However, the ADC is calculated based on the similar mathematical logic. 
+"""
+
+def calculate_aac(signal: np.ndarray, X: int) -> float:
+    """
+    Calculate the Acceleration Capacity (AAC) of a signal.
+
+    The formula for AAC is given by:
+    AAC = (1/X) * Σ(signal[i]) from i=0 to X-1 - (1/X) * Σ(signal[i]) from i=-X to -1
+
+    This calculates the difference between the mean of the signal values 
+    immediately after the anchor point (up to X-1) and the mean of the signal 
+    values immediately before the anchor point (up to X back from the anchor point).
+
+    Parameters:
+    signal (np.ndarray): The input signal, where the anchor point is at index X.
+    X (int): The anchor point index in the signal around which AAC is calculated.
+
+    Returns:
+    float: The calculated AAC value.
+    """
+    mean_after = np.mean(signal[X:X+X]) if X + X <= len(signal) else np.nan
+    mean_before = np.mean(signal[max(X-X, 0):X])
+    
+    aac = mean_after - mean_before
+
+    return aac
+
+def calculate_adc(signal: np.ndarray, X: int) -> float:
+    """
+    Calculate the Deceleration Capacity (ADC) of a signal, assuming a symmetrical definition to AAC.
```

Comment:
Those names are misleading: It should be rather clear- accelartion (AC), deccelartion (DC), anchors etc.